components/strava/actions/list-activities/list-activities.mjs

```diff
@@ -0,0 +1,47 @@
+import strava from "../../strava.app.js";
+
+export default {
+  name: "Get Activity List",
+  description: "Returns the activities of an athlete for a specific identifier. Requires activity:read. Only Me activities will be filtered out unless requested by a token with activity:read_all. See `https://developers.strava.com/docs/reference/`",
+  key: "strava-get-activity-list",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    strava,
+    before: {
+      type: "integer",
+      label: "Before",
+      description: "An epoch timestamp to use for filtering activities that have taken place before a certain time",
+      optional: true,
+    },
+    after: {
+      type: "integer",
+      label: "After",
+      description: "An epoch timestamp to use for filtering activities that have taken place after a certain time",
+      optional: true,
+    },
+    page: {
```

Comment:
We actually recommend handling pagination within the action / method, and simply fetching all records given the other criteria (e.g. `before` / `after`). This way a user doesn't have to ever think about pagination themselves - it's just handled by Pipedream.

Then, you can set an optional prop called `maxItems` that specifies the maximum number of items the action fetches. This should not have a default value (i.e. it should be `undefined`), so that the action fetches _all_ items from the API endpoint by default. Then if the user wants to list only the first N items, they can optionally set `maxItems` to N.

Reply:
In Strava's docs(https://developers.strava.com/docs/reference/#api-Activities-getLoggedInAthleteActivities) it is written that the default page size is 30. So as I understand, if we don't provide `per_page` field, the API will return first page with size 30.
In this situation, to handle this kind of paging, should we call the API with something like per_page=INFINITY?

Reply:
Sorry, I missed this. In general I do this:

1. Find out what the max page size is for the given endpoint
2. Write the app file method to fetch pages of that max size, until there are no more records. Return _all_ items across all pages to the user.
3. If the user sets the `maxItems` prop, your method should paginate until you find `maxItems` items, then return those items.

Users don't care about pagination and they explicitly don't want to handle it, so if we handle it for them and just return all results back, normally that's what they want. If they want to limit the result set, they can always set `maxItems`. And if they really want to handle pagination themselves, they can always write that code manually. But for actions, we like to abstract it.

Does that help clarify?